client/components/command-palette/test/index.tsx

```diff
@@ -0,0 +1,183 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { act, fireEvent, render, screen, waitFor } from '@testing-library/react';
+import React from 'react';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import { getCurrentRoutePattern } from '../../../state/selectors/get-current-route-pattern';
+import CommandPalette from '../index';
+import { useCommandPalette } from '../use-command-palette';
+
+const INITIAL_STATE = {
+	sites: {
+		items: {},
+	},
+	currentUser: {
+		capabilities: {
+			[ 1 ]: {
+				manage_options: true,
+			},
+		},
+	},
+	ui: {
+		selectedSiteId: 1,
+	},
+};
+
+const commands = [
+	{
+		name: 'getHelp',
+		label: 'Get help',
+	},
+	{
+		name: 'sendFeedback',
+		label: 'Send feedback',
+		searchLabel: 'find help',
+	},
+	{
+		name: 'clearCache',
+		label: 'Clear cache',
+		callback: jest.fn(),
+	},
+	{
+		name: 'enableEdgeCache',
+		label: 'Enable edge cache',
+	},
+];
+const mockStore = configureStore();
+const store = mockStore( INITIAL_STATE );
+
+jest.mock( '../../../state/selectors/get-current-route-pattern' );
+jest.mock( '../use-command-palette' );
+
+describe( 'CommandPalette', () => {
+	( getCurrentRoutePattern as jest.Mock ).mockReturnValue( '/sites' );
+
+	const renderCommandPalette = () => {
+		( useCommandPalette as jest.Mock ).mockReturnValue( {
+			commands: commands,
+			filterNotice: 'Mock Filter Notice',
+			emptyListNotice: 'Mock Empty List Notice',
+		} );
+
+		act( () => {
+			fireEvent.keyDown( document, { key: 'k', metaKey: true } );
+		} );
+
+		render(
+			<Provider store={ store }>
+				<CommandPalette />
+			</Provider>
+		);
+	};
+
+	it( 'should confirm that the command palette opens with the commands from the commands array', () => {
+		renderCommandPalette();
+
+		waitFor( () => {
+			expect( screen.getByPlaceholderText( 'Search for commands' ) ).toBeInTheDocument();
```

Comment:
It's considered a good practice to use `data-testid` attribute instead of text to test components.

#### Why?
Because of the separation of concerns of presentation and logic. If the text changes, those tests will be broken. However this in reality would have nothing to do with the tests. It's just the text that changed.

Take a look [here](https://testing-library.com/docs/queries/bytestid/) and in [react testing library priority section](https://testing-library.com/docs/queries/about/#priority)

> Test IDs
getByTestId: The user cannot see (or hear) these, so this is only recommended for cases where you can't match by role or text or it doesn't make sense (e.g. **the text is dynamic**).

It's not a hard requirement, but a good thing to do.
What do you think?

Reply:
I don't have a strong preference as long as the strings live in the same codebase. If we modify the placeholder  the test will fail and we can update the test. I agree that those strings are not expected to change.